### ISSUE TITLE
Fused softmax with mlir lowering is slower than non-mlir

### ISSUE THREAD

**Issue by Tixxx** [google/jax#13742]
### Description

Fused softmax kernel with mlir lowering is slower than non-fused version especially for large tensors.
Using the repro script below, setting flags `xla_gpu_enable_softmax_fusion` and `xla_gpu_enable_mlir_lowering` to false will speed up the softmax by around 30%.
Attached HLO for mlir and non-mlir cases.

With mlir, execution time per iteration is not consistent, using the program below, each iteration of non-mlir case is always between 900us-1ms while mlir varies from 1ms - 3ms.

Also based on the profiling results, mlir doesn't produce the most optimal configurations for kernel launch.
i.e. The single custom call in mlir version launches kernels with these params:
`grid size - 1000, block size - 256, threads - 256000`
while non-mlir version launches the fusion block with sum reduction using these params:
`grid size - 8000, block size - 384, threads - 3072000`.
This results in mlir kernels under-utilizing GPUs. The total compute utilization is almost 50% less than non-mlir version.

```python
import jax
import os
import jax
import jax.numpy as jnp
import datetime

os.environ['XLA_FLAGS'] = "--xla_gpu_enable_softmax_fusion=true --xla_gpu_enable_mlir_lowering=true"

def softmax_test(a):
    return jax.nn.softmax(a)

A = jnp.ones((20, 400, 6000))

start = datetime.datetime.now()
for _ in range(10000):
    jax.jit(softmax_test)(A)
end = datetime.datetime.now()
total = round((end-start).total_seconds()*1000)
print("Total: {} ms".format(total))
```

mlir fused case:
[module_0002.jit_softmax_test.sm_8.6_gpu_after_optimizations.txt](https://github.com/google/jax/files/10274570/module_0002.jit_softmax_test.sm_8.6_gpu_after_optimizations.txt)

non-mlir case:
[module_0002.jit_softmax_test.sm_8.6_gpu_after_optimizations.txt](https://github.com/google/jax/files/10274571/module_0002.jit_softmax_test.sm_8.6_gpu_after_optimizations.txt)

### What jax/jaxlib version are you using?

jax 0.4.1, jaxlib 0.4.1+cuda11.cudnn86

### Which accelerator(s) are you using?

GPU

### Additional system info

python 3.9.12, ubuntu 22.04

### NVIDIA GPU info

```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 515.43.04    Driver Version: 515.43.04    CUDA Version: 11.7     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  On   | 00000000:02:00.0  On |                  N/A |
| 35%   36C    P3    87W / 350W |    833MiB / 12288MiB |     10%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
```

cc: @cheshire  @jprabhas

**Comment by akuegel** [comment]
Thanks for filing this issue. We could reproduce the slowdown, and it seems to happen if the reduction dimension has > 256 elements. For now, we will disable using the new pipeline for these cases.

**Comment by mattjj** [comment]
Thanks @akuegel! Can you let us know when that fix is in? We'll need to update jaxlib.

**Comment by akuegel** [comment]
> Thanks @akuegel! Can you let us know when that fix is in? We'll need to update jaxlib.

It hast already landed in Tensorflow repository.
